src/app/layout.tsx

```diff
@@ -20,10 +21,27 @@ export default async function RootLayout({
   const apiClient = new BackendApiClient(headersObject);
 
   let user: ZetkinUser | null;
+  let session: ZetkinSession | null;
+
   try {
+    session = await apiClient.get<ZetkinSession>('/api/session');
     user = await apiClient.get<ZetkinUser>('/api/users/me');
   } catch (e) {
     user = null;
+    session = null;
+  }
+
+  // Check if user is logged in but not properly verified based on auth factors
+  const path = headersList.get('x-requested-path') || '';
+
+  if (user && session && session.factors && !path.startsWith('/verify')) {
+    const hasEmailAuth = session.factors.includes('email_password');
+
+    // Only redirect to email verification if user used email/password
+    // authentication and their email is not verified.
+    if (hasEmailAuth && !user.email_is_verified) {
+      redirect('/verify');
+    }
```

Comment:
This is already done in `redirectIfLoginNeeded()`, and I'm not sure we need to do it here as well?